tensorflow/python/kernel_tests/conv2d_transpose_test.py

```diff
@@ -157,6 +158,129 @@ def testGradient(self):
     err_tolerance = 0.0005
     self.assertLess(err, err_tolerance)
 
+  def get_available_gpus(self):
+    local_device_protos = device_lib.list_local_devices()
+    return [x.name for x in local_device_protos if x.device_type == 'GPU']
+
+  def testConv2DTransposeSingleStrideNCHW(self):
+    # `NCHW` data fomat is only supported for `GPU` device.
+    if len(self.get_available_gpus()) < 1:
```

Comment:
Could you use is_gpu_available() and do something like this? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/bias_op_test.py#L92

Note that you can use self.test_session(use_gpu=use_gpu) to put the op on GPU.